src/rpc/methods/eth/filter/mod.rs

```diff
@@ -71,18 +72,50 @@ impl EthEventHandler {
                 .context("Installation error")?;
 
             if let Some(filter_store) = &self.filter_store {
-                if filter_store.add(filter.clone()).is_err() {
-                    if let Some(tipset_filter_manager) = &self.tipset_filter_manager {
-                        let _ = tipset_filter_manager.remove(filter.id());
+                if let Err(err) = filter_store.add(filter.clone()) {
+                    if let Some(event_filter_manager) = &self.event_filter_manager {
+                        ensure!(
+                            event_filter_manager.remove(filter.id()).is_some(),
+                            "Filter not found"
+                        );
                     }
-                    return Err(Error::msg("Removal error"));
+                    bail!("Adding filter failed: {}", err);
                 }
             }
             Ok(filter.id().clone())
         } else {
             Err(Error::msg("NotSupported"))
         }
     }
+
+    // Installs an eth tipset filter
+    pub fn eth_new_block_filter(&self) -> Result<FilterID, Error> {
+        ensure!(
+            self.filter_store.is_some() && self.tipset_filter_manager.is_some(),
+            "NotSupported"
+        );
+
+        let tipset_manager = self
+            .tipset_filter_manager
+            .as_ref()
+            .context("Tipset filter manager not found")?;
+
+        let filter = tipset_manager.install().context("Installation error")?;
+
+        if let Some(filter_store) = &self.filter_store {
```

Comment:
We ensured the filter store is there in the first line of the method, correct? Is there a possibility it will be removed in the meantime? Same goes for tipset filter manager. Is there a need to check if those are null given the prior checks?